### PR TITLE
JS: Add support for threat models

### PR DESCRIPTION
**Pull request by RasmusWL**
This PR mostly follows the structure of the Python PR (https://github.com/github/codeql/pull/17203)

The most interesting aspect of this PR is that in https://github.com/github/codeql/pull/17256/commits/f733ac19a9dc1c8df5c4672695a84006101f2969 I was only able to meaningfully add ActiveThreatModelSource as the default source for some queries, since some of them rely on this pattern:

```codeql
class RemoteFlowSourceAsSource extends Source instanceof RemoteFlowSource {
  RemoteFlowSourceAsSource() { not this instanceof ClientSideRemoteFlowSource }
}
```

and it's not quite clear how these should be migrated :thinking: This pattern is used in these 7 cases:

```
javascript/ql/lib/semmle/javascript/security/dataflow/ClientSideUrlRedirectCustomizations.qll
javascript/ql/lib/semmle/javascript/security/dataflow/CommandInjectionCustomizations.qll
javascript/ql/lib/semmle/javascript/security/dataflow/CorsMisconfigurationForCredentialsCustomizations.qll
javascript/ql/lib/semmle/javascript/security/dataflow/RegExpInjectionCustomizations.qll
javascript/ql/lib/semmle/javascript/security/dataflow/RequestForgeryCustomizations.qll
javascript/ql/lib/semmle/javascript/security/dataflow/ResourceExhaustionCustomizations.qll
javascript/ql/lib/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll
```

I'll also note that while I think the support for `environment`, `commandargs`, `database` threat-models are OK, the ones for `stdin` and `file` are a little simple right now. (see 3 last commit messages for more context)

### CODE REVIEW

**Review by asgerf — COMMENTED** [review]
The changes to JS look good to me so far 👍

File: javascript/ql/lib/ext/default-threat-models-fixup.model.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+extensions:
+  - addsTo:
+      pack: codeql/threat-models
+      extensible: threatModelConfiguration
+    data:
+    # Since responses are enabled by default in the shared threat-models configuration,
+    # we need to disable it here to keep existing behavior for the python analysis.
```
[the remark this answers is not in the document]
Comment (RasmusWL) [reply javascript/ql/lib/ext/default-threat-models-fixup.model.yml]: :+1: Done as part of rebase just now

**Comment by RasmusWL** [comment]
Have added proper docs integration, and have provided some modeling for database/stdin/file (although the latter two could be improved more). Ready for review now, will do DCA soon.

File: javascript/ql/lib/semmle/javascript/frameworks/NodeJSLib.model.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+extensions:
+  # Make sure that the extensible model predicates have at least one definition
+  # to avoid errors about undefined extensionals.
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/frameworks/NodeJSLib.model.yml:3]:
```suggestion
```

**Comment by asgerf** [comment]
Looks good so far, but will take another look tomorrow when I have more time.

> The most interesting aspect of this PR is that in https://github.com/github/codeql/commit/f733ac19a9dc1c8df5c4672695a84006101f2969 I was only able to meaningfully add ActiveThreatModelSource as the default source for some queries, since some of them rely on this pattern:
```codeql
class RemoteFlowSourceAsSource extends Source instanceof RemoteFlowSource {
  RemoteFlowSourceAsSource() { not this instanceof ClientSideRemoteFlowSource }
}
```

Just change `RemoteFlowSource` to `ActiveThreatModelSource` while keeping the `not` in the body.

**Comment by RasmusWL** [comment]
> Just change `RemoteFlowSource` to `ActiveThreatModelSource` while keeping the `not` in the body.

:exploding_head: so obvious after you've seen it :joy:

**Review by asgerf — APPROVED** [review]
👍

**Pull request merged**